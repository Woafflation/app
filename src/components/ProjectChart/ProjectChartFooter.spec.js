/* eslint-env jest */
import React from 'react'
import { LinkToSocialTool } from './ProjectChartFooter'
import toJson from 'enzyme-to-json'
import { mount, shallow } from 'enzyme'

describe('ProjectChartFooter', () => {
  it('LinkToSocialTool', () => {
    const project = {
      name: 'Santiment Network Token',
      ticker: 'SAN',
      slug: 'santiment'
    }
    const wrapper = shallow(<LinkToSocialTool {...project} />)
    expect(wrapper.props().to).toEqual(
      '/labs/trends/explore/santiment%20OR%20SAN%20OR%20Santiment%20Network%20Token'
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it("LinkToSocialTool, if project doesn't have name", () => {
    const project = {
      name: null,
      ticker: 'SAN',
      slug: 'santiment'
    }
    const wrapper = shallow(<LinkToSocialTool {...project} />)
    expect(wrapper.props().to).toEqual(
      '/labs/trends/explore/santiment%20OR%20SAN'
    )

    const projectWithoutTicker = {
      name: null,
      ticker: null,
      slug: 'santiment'
    }
    const wrapper2 = shallow(<LinkToSocialTool {...projectWithoutTicker} />)
    expect(wrapper2.props().to).toEqual('/labs/trends/explore/santiment')
  })
})
