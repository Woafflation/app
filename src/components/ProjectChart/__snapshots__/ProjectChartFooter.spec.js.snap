// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectChartFooter LinkToSocialTool 1`] = `
<Button
  accent="positive"
  as={[Function]}
  border={false}
  className=""
  fluid={false}
  isActive={false}
  to="/labs/trends/explore/santiment%20OR%20SAN%20OR%20Santiment%20Network%20Token"
  variant="fill"
>
  Social analyze
</Button>
`;
